core/arch/arm/plat-imx/imx_caam.c

```diff
@@ -13,7 +13,7 @@
 
 void init_caam(void)
 {
-	struct imx_caam_ctrl *caam = (struct caam_ctrl *)CAAM_BASE;
+	struct imx_caam_ctrl *caam = (struct imx_caam_ctrl *)CAAM_BASE;
```

Comment:
If this is ever compiled for a 64-bit platform you may need to insert a `(vaddr_t)` before the final cast.